### ISSUE TITLE
Domoticz on Windows - Shall it work?

### ISSUE THREAD

**Issue by RicardP** [Xorfor/Domoticz-PWS-Plugin#1]
Hi, exited to find this plugin as I purchased a Weather Station using EasyWeather 1.4.6 app. 
So I installed your package under plugins and all devices was created, so far so good... however I do not get any values updated and some of the Icons is missing in on the created Devices.

I double and triple checked every setting to your guide but no success getting data into Domoticz.

Thanks for your nice work!

//Ricard

**Comment by Xorfor** [comment]
There should be no difference between Windows or Unix. 

Do you have other Domoticz Plugins on your system?
Can you set Debug on True and give me the log?

**Comment by RicardP** [comment]
Thank you, maybe not so much but here is from Domoticz status log...

 2020-01-17 16:35:23.671 Status: (Väderstation) Stop directive received.
2020-01-17 16:35:23.671 Status: (Väderstation) Exiting work loop.
2020-01-17 16:35:23.890 Status: (Väderstation) Stopping threads.
2020-01-17 16:35:23.890 Status: (Väderstation) Stopped.
2020-01-17 16:35:23.890 Status: (Väderstation) Started.
2020-01-17 16:35:24.359 (Väderstation) Debug logging mask set to: PYTHON PLUGIN QUEUE IMAGE DEVICE CONNECTION MESSAGE ALL
2020-01-17 16:35:24.359 (Väderstation) onStart
2020-01-17 16:35:24.359 (Väderstation) Pushing 'ProtocolDirective' on to queue
2020-01-17 16:35:24.359 (Väderstation) Pushing 'ListenDirective' on to queue
2020-01-17 16:35:24.359 (Väderstation) Listening to port: 2244
2020-01-17 16:35:24.359 (Väderstation) Processing 'ProtocolDirective' message
2020-01-17 16:35:24.359 (Väderstation) Protocol set to: 'HTTP'.
2020-01-17 16:35:24.359 (Väderstation) Processing 'ListenDirective' message
2020-01-17 16:35:24.359 (Väderstation) Transport set to: 'TCP/IP', :2244.
2020-01-17 16:35:24.359 (Väderstation) Listen directive received, action initiated successfully.
2020-01-17 16:35:24.343 Status: (Väderstation) Initialized version 1.0.7, author 'Xorfor'
2020-01-17 16:35:24.343 Status: (Väderstation) Entering work loop.
2020-01-17 16:35:34.593 (Väderstation) Pushing 'onHeartbeatCallback' on to queue
2020-01-17 16:35:34.609 (Väderstation) Processing 'onHeartbeatCallback' message
2020-01-17 16:35:34.609 (Väderstation) Calling message handler 'onHeartbeat'.
2020-01-17 16:35:34.609 (Väderstation) onHeartbeat
2020-01-17 16:35:44.827 (Väderstation) Pushing 'onHeartbeatCallback' on to queue
2020-01-17 16:35:44.858 (Väderstation) Processing 'onHeartbeatCallback' message
2020-01-17 16:35:44.858 (Väderstation) Calling message handler 'onHeartbeat'.
2020-01-17 16:35:44.858 (Väderstation) onHeartbeat
2020-01-17 16:35:55.062 (Väderstation) Pushing 'onHeartbeatCallback' on to queue
2020-01-17 16:35:55.108 (Väderstation) Processing 'onHeartbeatCallback' message
2020-01-17 16:35:55.108 (Väderstation) Calling message handler 'onHeartbeat'.
2020-01-17 16:35:55.108 (Väderstation) onHeartbeat

And error Log from yesterday test...

 2020-01-17 00:02:27.625 Error: EventSystem: TEMP_HUM_BARO missing values : ID=2070, sValue=
2020-01-17 00:02:36.843 Error: EventSystem: TEMP_HUM_BARO missing values : ID=2070, sValue=
2020-01-17 00:02:41.843 Error: EventSystem: TEMP_HUM_BARO missing values : ID=2070, sValue=
2020-01-17 00:02:47.437 Error: EventSystem: TEMP_HUM_BARO missing values : ID=2070, sValue=

...and repeating

I am not using any other Python plugin at the moment but some folders present, not active.

I did not have any general issues with Python plugins before, but at least one tested did not work but it was stated "plugin was not for windows platform". But I tested it anyway because as you say Python code should/could work generally on its supported platforms.

**Comment by Xorfor** [comment]
This is not the log I expected. I expect to see something like:

OnMessage: ...
Request: ...
strData: ...

Do you get that kind of information? If not, are you sure you have set the port in WS View on 2244?

**Comment by RicardP** [comment]
Yes I did set WS View to 2244, first 5000 as default but tested 2244 just to try if there were some kind of conflict on 5000.
However now I notice that the port number assigned in Domoticz plugin setting is entered into address field not port!?

![clip](https://user-images.githubusercontent.com/21274628/72662884-85518a00-39ec-11ea-84b7-f957e7710a39.JPG)

**Comment by RicardP** [comment]
![Screenshot_20200118-122316](https://user-images.githubusercontent.com/21274628/72662967-71f2ee80-39ed-11ea-98c5-89c363cba82f.png)

**Comment by Xorfor** [comment]
I assume that 192.168.1.14 is also the ip address of your Domoticz server? Can you put back the port to 5000, both on in Domoticz and in WS View. It will take some minutes for a PWS to start sending the data.
Please send me the log starting with OnMessage: ...

**Comment by RicardP** [comment]
Hi, yes that is the Domoticz server IP, but 192.168.1.4 (not ending .14 that you mistakenly wrote). However, I do not get the OnMessage: in my log, or which log do you mean? :)

As you can see also some icons missing for some Devices...

![a](https://user-images.githubusercontent.com/21274628/72938084-9bec3e00-3d6a-11ea-98ba-a20cefd72255.JPG)

![b](https://user-images.githubusercontent.com/21274628/72938086-9db60180-3d6a-11ea-8cf1-837cc42dcd18.JPG)

![c](https://user-images.githubusercontent.com/21274628/72938092-a0185b80-3d6a-11ea-8cf7-bce9ec00a870.JPG)

**Comment by Xorfor** [comment]
If the plugin has Debug: True and you don't get a line in the Domoticz for this plugin with `...onMessage...`, then that means that your PWS is **not** sending its data to Domoticz. That also explains why your PWS devices in Domoticz doesn't have values and/or icons. These devices still have their initial settings and are not updated. 
Like the device `Station` doesn't display the PWS information! 

Please check if your PWS is connected on the same network as Domoticz. 
Also double check the ip and port settings, etc.

**Comment by RicardP** [comment]
Hi @Xorfor I checked everything over and over but all looks fine, even restarted WS display unit, I can see the PWS display listed with correct IP address assigned in the WS View as well... 

As said before the Port number found in Address table in hardware overview!? That is of course not correct but if it has something to do with my problem I don't know...??

![clip](https://user-images.githubusercontent.com/21274628/73119637-11236300-3f65-11ea-81b8-06b9d15240e7.JPG)

**Comment by Xorfor** [comment]
That the `5000` is shown in address is not the problem. I have the same and Address and Port are just settings for the hardware which are displayed on this page. Try the standard hardware `Winddelen` and you will see the name of the selected windmill.

I have no idea yet what the problem is. I have to setup a windows Domoticz server to be able to debug this.

**Comment by Xorfor** [comment]
I did setup a windows 10 server and it works as expected!!!

One thing: 

Did you specify: Setup > Settings > System > Local Networks (no username/password) > Networks: 127.0.0.*;192.168.1.*?

Perhaps that will help!

**Comment by RicardP** [comment]
@Xorfor Thanks a lot for your efforts, I got it working by another simple solution... turning of Windows Firewall. 
Even that I previously opened the Inbound Ports 5000 and second tested 2244 in Firewall access table it did not let those ports pass. Tried both UDP and TCP as well... 
I will test some more with activated Firewall and Ports settings to try find out what I did wrong. But if not successful I do really not need the Firewall activated in my private network.
